lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -170,6 +174,24 @@ module.exports = {
       ]);
     }
 
+    if (disableLogsFunctions.length > 0) {
+      // add deny rule for disabled logs function
+      const arnOfdisableLogGroups = disableLogsFunctions.map(functionObject => {
+        return {
+          'Fn::Sub':
+            'arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}' +
+            `:log-group:${this.provider.naming.getLogGroupName(functionObject.name)}:*`,
+        };
+      });
+      this.serverless.service.provider.compiledCloudFormationTemplate.Resources[
+        this.provider.naming.getRoleLogicalId()
+      ].Properties.Policies[0].PolicyDocument.Statement.push({
+        Effect: 'Deny',
```

Comment:
Wouldn't it be good enough to just not add `'Allow'` policy? I believe that's all needed (by default access is denied)

Reply:
@medikoo As serverless by default allow logs, so I think if we make by default access is denied so this makes more complicate the scenario. Let me know what u think?

Reply:
@as19ish What I meant that we should disable access simply by removing "Allow" rules, and not by by replacing them with "Deny" rules (they're effect-less, as they do no override anything)

Reply:
@medikoo 

I think there is some misunderstanding, I m explaining the whole problem.

By default, we are allowing lambda execution role to this permission

1. logs:CreateLogStream
2. logs:CreateLogGroup
3. logs:PutLogEvents

Note: **BY USING WILDCARD CHARACTER**

```
"Statement": [
        {
            "Action": [
                "logs:CreateLogStream",
                "logs:CreateLogGroup"
            ],
            "Resource": [
                "arn:aws:logs:us-east-1:188267231256:log-group:/aws/lambda/serverless-dev*:*"
            ],
            "Effect": "Allow"
        },
        {
            "Action": [
                "logs:PutLogEvents"
            ],
            "Resource": [
                "arn:aws:logs:us-east-1:188267231256:log-group:/aws/lambda/serverless-dev*:*:*"
            ],
            "Effect": "Allow"
        }
    ]
```

Now the main problem here how can we deny to specific lambda to do not put logs and also by default all lambda has the same lambda execution role.

So here what i'm doing is adding a new statement that denies for put logs for a specific function (full ARN not using a wildcard) as per the Principle of Least Privilege

```
{
            "Action": "logs:PutLogEvents",
            "Resource": [
                "arn:aws:logs:us-east-1:188267231256:log-group:/aws/lambda/serverless-dev-hello:*"
            ],
            "Effect": "Deny"
}
```

I m a little confused here. Can we do it in a better way?

Reply:
@as19ish ah, indeed, thanks for explanations. I forgot that in common cases we use wildcards, and indeed for this, we'd need to _deny_ the assigned rights.

Still it'll be good to integrate it better with logic which generates "Allow" rights, e.g. we should not create "Deny" rule, for functions not expressed by canonical naming (we should just prevent generation of "Allow" rights for them)

Therefore I would cover this in this part of logic: https://github.com/serverless/serverless/blob/6dd9a31161d57871663872cd458d8027ec496b81/lib/plugins/aws/package/lib/mergeIamTemplates.js#L92-L129

1. For non canonical named functions, do not create "Allow" rights
2. Group names of all disabled canonically named functions, and generate "Deny" rights in scope of this condition: https://github.com/serverless/serverless/blob/6dd9a31161d57871663872cd458d8027ec496b81/lib/plugins/aws/package/lib/mergeIamTemplates.js#L116-L129